### ISSUE TITLE
Loss converged but very high loss/ low mAP

### ISSUE THREAD

**Issue by avocalex** [AlexeyAB/darknet#3564]
I am working on object detection using a large scale autonomous dataset (BDD100K). My end goal is to detect common road objects using a Jetson TX2. Currently, I'm training this dataset on 2x1080Ti GPUs. I've been using yolov3-tiny_obj.cfg file to train this dataset but most of my training results in mAP <30% and loss >10. Here are the things I've attempted so far:

- Tried different hyperparameter settings (pretty much anything I could find in the darknet/issues); changed width/height, learning rate, burn_in, max_batches, Adam, etc. Currently training on width/height = 832 (will update if this improves result). 
- Recalculated the anchors. The results below used default anchors from tiny yolo.
- Added max=100 on [YOLO] layers
- Set random = 1
- Checked labels using yolo_mark
- Train for at least 20,000 iterations

 I've tried many different configurations of the CFG file but none of the results seem to go below 10 even though they converge. Just to clarify I am using Alexeyab's darknet. The following is the best result I got so far. I would appreciate any inputs on how I can decrease loss/increase mAP.

Here are my configurations:

**Makefile**
GPU=1
CUDNN=1
OPENCV=1

**CFG**

batch=64
subdivisions=16
width=736
height=416
channels=3
momentum=0.9
decay=0.0005
angle=0
saturation = 1.5
exposure = 1.5
hue=.1
 
learning_rate=0.0005
burn_in=4000
max_batches = 40000
policy=steps
steps=32000,36000
scales=.1,.1

**Result**
_Loss_
![7_7](https://user-images.githubusercontent.com/48293538/60770674-a7528180-a118-11e9-91a3-21adb66ded42.jpg)

_mAP_
detections_count = 1890545, unique_truth_count = 231561  
class_id = 0, name = bike, ap = 14.78%   	 (TP = 130, FP = 146) 
class_id = 1, name = bus, ap = 32.39%   	 (TP = 570, FP = 445) 
class_id = 2, name = car, ap = 51.59%   	 (TP = 68393, FP = 49197) 
class_id = 3, name = motor, ap = 7.17%   	 (TP = 6, FP = 9) 
class_id = 4, name = person, ap = 21.40%   	 (TP = 4229, FP = 5724) 
class_id = 5, name = rider, ap = 14.20%   	 (TP = 92, FP = 85) 
class_id = 6, name = traffic light, ap = 23.43%   	 (TP = 12605, FP = 21143) 
class_id = 7, name = traffic sign, ap = 28.61%   	 (TP = 15355, FP = 20268) 
class_id = 8, name = train, ap = 0.00%   	 (TP = 0, FP = 0) 
class_id = 9, name = truck, ap = 35.00%   	 (TP = 1627, FP = 1070) 

 for thresh = 0.25, precision = 0.51, recall = 0.44, F1-score = 0.48 
 for thresh = 0.25, TP = 103007, FP = 98087, FN = 128554, average IoU = 36.79 % 

 IoU threshold = 50 %, used Area-Under-Curve for each unique Recall 
 mean average precision (mAP@0.50) = 0.228573, or 22.86 % 
Total Detection Time: 83.000000 Seconds

*mAP is slightly low due to missing images of "train" class but overall very low mAP. 

**Anchors**
for  736x416
anchors = 9,9 20,14, 13,25, 36,26, 25,54, 65,45, 100,85, 170,128, 247,220
for  832x832
anchors = 10,18, 22,27, 14,50, 40,52, 28,108, 74,90, 113,169, 192,257, 279,439

**Comment by AlexeyAB** [comment]
>  I've been using yolov3-tiny_obj.cfg file to train this dataset but most of my training results in mAP <30% and loss >10. 

Loss doesn't matter. You should look at mAP.


> for 736x416
> anchors = 9,9 20,14, 13,25, 36,26, 25,54, 65,45, 100,85, 170,128, 247,220
> for 832x832
> anchors = 10,18, 22,27, 14,50, 40,52, 28,108, 74,90, 113,169, 192,257, 279,439

Are these anchors calculated on your dataset?

Will you use this repository for the final detection, or will you use other repos?

**Comment by avocalex** [comment]
Thanks for your reply, I appreciate it.

> Are these anchors calculated on your dataset?

Yes, they were calculated from the dataset. 

> Will you use this repository for the final detection, or will you use other repos? If you will use this repo - then you can try to use one of these models: [#3114 (comment)]

I will use this repo for the final detection. I've tried yolo_v3_tiny_pan_lstm.cfg and got the following results: 

![chart](https://user-images.githubusercontent.com/48293538/60787174-e671e880-a193-11e9-81d2-2b7d8677eb83.png)

![7_9](https://user-images.githubusercontent.com/48293538/60787180-ed006000-a193-11e9-9e84-79f2d62c583b.jpg)

detections_count = 2170232, unique_truth_count = 231561  
class_id = 0, name = bike, ap = 4.08%   	 (TP = 1, FP = 4) 
class_id = 1, name = bus, ap = 14.61%   	 (TP = 154, FP = 218) 
class_id = 2, name = car, ap = 50.95%   	 (TP = 61670, FP = 36993) 
class_id = 3, name = motor, ap = 3.10%   	 (TP = 0, FP = 0) 
class_id = 4, name = person, ap = 18.44%   	 (TP = 2265, FP = 2062) 
class_id = 5, name = rider, ap = 3.76%   	 (TP = 1, FP = 1) 
class_id = 6, name = traffic light, ap = 24.45%   	 (TP = 12217, FP = 18922) 
class_id = 7, name = traffic sign, ap = 31.88%   	 (TP = 13272, FP = 10995) 
class_id = 8, name = train, ap = 0.00%   	 (TP = 0, FP = 0) 
class_id = 9, name = truck, ap = 19.22%   	 (TP = 602, FP = 577) 

 for thresh = 0.25, precision = 0.56, recall = 0.39, F1-score = 0.46 
 for thresh = 0.25, TP = 90182, FP = 69772, FN = 141379, average IoU = 39.52 % 

 IoU threshold = 50 %, used Area-Under-Curve for each unique Recall 
 mean average precision (mAP@0.50) = 0.170488, or 17.05 % 
Total Detection Time: 160.000000 Seconds

---------------------------------------------------------------------------------------------------
I will try to train with other models but it seems to be that my mAP results are generally lower than what others are able to achieve. With the latest training, mAP decreased at one point and increased once again near the end of the training. Should increase batch size and train longer? 

*I currently have 57,500 images for training and 12,500 for validation

**Comment by LukeAI** [comment]
@avocalex you know the LSTM model requires the frames to be in a sequence from a video? bdd100k is in a random order not from a contiguous video stream.
It is quite a challenging dataset so you probably won't break through 50% with Yolo tiny. I managed to train tiny-yolo_3l to around 44% AP on bdd100k - I got better results with the original anchors from coco, not generated from the bdd dataset and better results with height :width set to approx 9 : 16. Also try lowering subdivisions to 4.
